cleanlab/outlier.py

```diff
@@ -0,0 +1,541 @@
+          You should never provide predictions on the same examples used to train the model,
+          as these will be overfit and unsuitable for finding label-errors.
+          To obtain out-of-sample predicted probabilities for every datapoint in your dataset, you can use :ref:`cross-validation <pred_probs_cross_val>`.
+          Alternatively it is ok if your model was trained on a separate dataset and you are only evaluating
+          data that was previously held-out.
+
+        labels : np.ndarray, optional
+          A discrete vector of noisy labels, i.e. some labels may be erroneous of shape ``(N,)``.
+          *Format requirements*: for dataset with K classes, labels must be in 0, 1, ..., K-1.
+          All the classes (0, 1, ..., and K-1) MUST be present in ``labels``, such that: ``len(set(labels)) == pred_probs.shape[1]``
+          Note: multi-label classification is not supported by this method, each example must belong to a single class, e.g. format: ``labels = np.ndarray([1,0,2,1,1,0...])``.
+
+        params : dict, default = None
+          Optional keyword arguments to that change how estimator is fit. Types of arguments passed depends on if `OutOfDistribution` object is fit on `features` or `pred_probs`.
+
+          If `features` is passed in, `params` could contain following keys:
+            *  k : int, default=None
+                  Optional number of neighbors to use when calculating outlier score (average distance to neighbors).
+                  If `k` is not provided, then by default ``k = knn.n_neighbors`` or ``k = 10`` if ``knn is None``.
+                  If an existing ``knn`` object is provided, you can still specify that outlier scores should use
+                  a different value of `k` than originally used in the ``knn``,
+                  as long as your specified value of `k` is smaller than the value originally used in ``knn``.
+            *  t : int, default=1
+                  Optional hyperparameter only for advanced users.
+                  Controls transformation of distances between examples into similarity scores that lie in [0,1].
+                  The transformation applied to distances `x` is `exp(-x*t)`.
+                  If you find your scores are all too close to 1, consider increasing `t`,
+                  although the relative scores of examples will still have the same ranking across the dataset.
+
+          If `pred_probs` is passed in, `params` could contian following keys:
+            *  adjust_pred_probs : bool, True
+                  Account for class imbalance in the label-quality scoring by adjusting predicted probabilities
+                  via subtraction of class confident thresholds and renormalization.
+                  Set this to ``False`` if you prefer to skip accounting for class-imbalance.
+                  See `Northcutt et al., 2021 <https://jair.org/index.php/jair/article/view/12125>`_.
```

Comment:
keep track of TODO to replace this with link to our writeup describing this methodology.